src/lexer.py

```diff
@@ -370,25 +542,36 @@ def lex(source: str) -> datatypes.TokenStream:
             location = get_srcloc(line, line_idx, start_idx, idx)
             tokens.append(datatypes.Token(type=datatypes.TokenType.Comma, data=None, location=location))
 
-        elif current_char == '.':
-            idx += 1
-
-            location = get_srcloc(line, line_idx, start_idx, idx)
-            tokens.append(datatypes.Token(type=datatypes.TokenType.Period, data=None, location=location))
-
         elif source[idx:idx+2] == "'s":
             idx += 2
 
             location = get_srcloc(line, line_idx, start_idx, idx)
             tokens.append(datatypes.Token(type=datatypes.TokenType.ReservedAssignment, data=None, location=location))
+            if not seen_keyword_on_this_line:
+                idx, token = tokenize_poetic_assignment(source, line, line_idx, idx)
+                tokens.append(token)
+            seen_keyword_on_this_line = True
 
         elif current_char.isalpha():
             idx, symbol = word_symbolizer(source, idx, error_func)
-
             contents = source[start_idx:idx]
             location = get_srcloc(line, line_idx, start_idx, idx)
             tokens.append(datatypes.Token(type=symbol, data=contents, location=location))
-
+            if not seen_keyword_on_this_line:
+                if symbol in POETIC_ASSIGNMENT_SYMBOLS:
+                    idx, token = tokenize_poetic_assignment(source, line, line_idx, idx)
+                    tokens.append(token)
+                    seen_keyword_on_this_line = True
+                elif symbol in POETIC_STRING_ASSIGNMENT_SYMBOLS:
+                    if source[idx] != " ":
```

Comment:
This should be `!source[idx].isspace()`

Reply:
I based this on the part of the spec that says `A poetic string literal assignment starts with a variable name, followed by one of the keywords says followed by a single space.` I agree that it would make sense to allow other space characters like tabs though. It would have to be `not source[idx].isspace() or source[idx] == "\n"` since skipping the newline character would result in it continuing the assignment on the next line.